src/Nox.Generator/Application/Commands/AddCommandGenerator.cs

```diff
@@ -23,11 +24,12 @@ public void Generate(SourceProductionContext context, NoxSolutionCodeGeneratorSt
         {
             context.CancellationToken.ThrowIfCancellationRequested();
 
-            var parent = codeGeneratorState.Solution.Domain.Entities.FirstOrDefault(e =>
-                e.OwnedRelationships?.Any(o => o.Entity == entity.Name && !o.WithSingleEntity) == true);
+            var parent = entity.TryGetParent(codeGeneratorState.Solution.Domain.Entities);
             if (parent is null)
                 continue;
 
+            var isSingleRelationship = parent.OwnedRelationships.First(o => o.Entity == entity.Name).WithSingleEntity;
```

Comment:
this is tricky because we are assuming an entity can only be owned by another and not by two for example

we need to prepare the code to generate a command for each parent. lets discuss this